Source/SwiftLintFramework/Models/Configuration.swift

```diff
@@ -47,6 +48,8 @@ public struct Configuration: Equatable {
     public let rules: [Rule]
     public var rootPath: String?              // the root path to search for nested configurations
     public var configurationPath: String?     // if successfully loaded from a path
+    public var hash: Int?
```

Comment:
A `Rule` is uniquely identified by its `identifier`, so its `hashValue` can just be `identifier.hashValue` and this can be implemented as a protocol extension.